### PR TITLE
Feature/resolve report

### PR DESCRIPTION
**Pull request by SharyuMarwadi**
Bug fixes, mail template fixes and admin panel code changes

### CODE REVIEW

File: internal/app/badges/service.go [hunk 1]
```diff
@@ -12,33 +15,54 @@ import (
...
+		userRepo:   userRepo,
 	}
 }
 
-func (bs *service) ListBadges(ctx context.Context) (resp []dto.Badge, err error) {
+func (bs *service) ListBadges(ctx context.Context) ([]dto.Badge, error) {
+
+	var resp []dto.Badge
+
+	fmt.Println("In list badges")
```
Comment (yama11299) [inline internal/app/badges/service.go:37]: remove testing logs from everywhere.

File: internal/app/badges/service.go [hunk 2]
```diff
@@ -12,33 +15,54 @@ import (
...
 	if err != nil {
 		logger.Error(err.Error())
 		err = apperrors.InternalServerError
+		return nil, err
 	}
 
 	for _, item := range dbResp {
-		svcItem := mapDbToSvc(item)
-		resp = append(resp, svcItem)
+		fmt.Println("updated by: ", item.UpdatedBy)
```
Comment (yama11299) [inline internal/app/badges/service.go:47]: -

File: internal/app/badges/service.go [hunk 3]
```diff
@@ -12,33 +15,54 @@ import (
...
+			user, err := bs.userRepo.GetUserById(ctx, reqData)
+			if err != nil {
+				return nil, err
+			}
+			svcItem := mapDbToSvc(item, user)
+			resp = append(resp, svcItem)
+		} else {
+			svcItem := mapDbToSvc(item, dto.GetUserByIdResp{})
+			resp = append(resp, svcItem)
+		}
```
Comment (yama11299) [inline internal/app/badges/service.go:62]: add comments for above logic, can explain the scenarios

File: internal/app/grades/service.go [hunk 4]
```diff
@@ -34,8 +39,22 @@ func (gs *service) ListGrades(ctx context.Context) (resp []dto.Grade, err error)
 	}
 
 	for _, item := range dbResp {
-		svcItem := mapDbToSvc(item)
-		resp = append(resp, svcItem)
+		fmt.Println("updated by: ", item.UpdatedBy)
```
Comment (yama11299) [inline internal/app/grades/service.go:42]: -

File: internal/app/grades/service.go [hunk 5]
```diff
@@ -64,9 +91,17 @@ func (gs *service) EditGrade(ctx context.Context, id string, points int64) (err
...
+	svcResp.UpdatedBy = user.FirstName + " " + user.LastName
 	return
 }
+
+func GetQuarterStartUnixTime() int64 {
+	// Example function to get the Unix timestamp of the start of the quarter
+	now := time.Now()
+	quarterStart := time.Date(now.Year(), (now.Month()-1)/3*3+1, 1, 0, 0, 0, 0, time.UTC)
+	return quarterStart.Unix() * 1000 // convert to milliseconds
+}
```
Comment (yama11299) [inline internal/app/grades/service.go:107]: remove this, already defined above.

File: internal/app/reportAppreciations/service.go [hunk 6]
```diff
@@ -230,16 +231,24 @@ func (rs *service) DeleteAppreciation(ctx context.Context, reqData dto.Moderatio
 		return
 	}
 
+	seconds := appreciation.CreatedAt / 1000
+	nanoseconds := (appreciation.CreatedAt % 1000) * 1e6
+
+	tm := time.Unix(seconds, nanoseconds)
+	formattedDate := tm.Format("02/01/2006")
```
Comment (yama11299) [inline internal/app/reportAppreciations/service.go:238]: comment the date format in dd/mm/yyyy or mm/dd/yyyy

File: internal/app/reportAppreciations/service.go [hunk 7]
```diff
@@ -370,36 +377,52 @@ func (rs *service) ResolveAppreciation(ctx context.Context, reqData dto.Moderati
...
+	mailReq := email.NewMail([]string{reporterEmail}, []string{}, []string{}, "Results of reported appreciation")
 	mailReq.ParseTemplate("./internal/app/email/templates/deleteAppreciation.html", templateData)
-	err := mailReq.Send(plainTextContent)
+	err := mailReq.Send()
+	if err != nil {
+		logger.Errorf("err: %v", err)
+		return err
+	}
+
+	logger.Info("sender email: ---------> ", reporterEmail)
```
Comment (yama11299) [inline internal/app/reportAppreciations/service.go:399]: ?

**Comment by SharyuMarwadi** [comment]
Changes done

**Comment by yama11299** [comment]
Approved.

**Pull request merged**